testutils/lint/passes/hash: Test failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/ca494e6d787a2fb516e3e1036a46917fbe4b8458

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=Test PKG=github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/hash TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1432782&tab=buildLog

```
=== RUN   Test
--- FAIL: Test (0.08s)
    analysistest.go:103: loading [a]: go [list -e -json -compiled=true -test=true -export=false -deps=true -find=false -- a]: exit status 1: go build math: failed to cache compiled Go files

```

SHA: https://github.com/cockroachdb/cockroach/commits/96a27d8fc28c7f32ea640a1473b16b74e6825588

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=Test PKG=github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/hash TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1434916&tab=buildLog

```
=== RUN   Test
--- FAIL: Test (0.08s)
    analysistest.go:103: loading [a]: go [list -e -json -compiled=true -test=true -export=false -deps=true -find=false -- a]: exit status 1: go build crypto/sha256: failed to cache compiled Go files

```

SHA: https://github.com/cockroachdb/cockroach/commits/bad5e2ea8674f0513f949d3940609b1f4d685581

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=Test PKG=github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/hash TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1438425&tab=buildLog

```
=== RUN   Test
--- FAIL: Test (0.10s)
    analysistest.go:103: loading [a]: go [list -e -json -compiled=true -test=true -export=false -deps=true -find=false -- a]: exit status 1: go build crypto/sha256: failed to cache compiled Go files

```

--- a/pkg/testutils/lint/passes/hash/hash_test.go
+++ b/pkg/testutils/lint/passes/hash/hash_test.go
@@ -13,11 +13,15 @@ package hash_test
 import (
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/hash"
 	"golang.org/x/tools/go/analysis/analysistest"
 )
 
 func Test(t *testing.T) {
+	if testutils.NightlyStress() {
+		t.Skip("Go cache files don't work under stress")
+	}
 	testdata := analysistest.TestData()
 	analysistest.Run(t, testdata, hash.Analyzer, "a")
 }

--- a/pkg/testutils/lint/passes/timer/timer_test.go
+++ b/pkg/testutils/lint/passes/timer/timer_test.go
@@ -13,11 +13,15 @@ package timer_test
 import (
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/timer"
 	"golang.org/x/tools/go/analysis/analysistest"
 )
 
 func Test(t *testing.T) {
+	if testutils.NightlyStress() {
+		t.Skip("Go cache files don't work under stress")
+	}
 	testdata := analysistest.TestData()
 	analysistest.Run(t, testdata, timer.Analyzer, "a")
 }

--- a/pkg/testutils/lint/passes/unconvert/unconvert_test.go
+++ b/pkg/testutils/lint/passes/unconvert/unconvert_test.go
@@ -13,11 +13,15 @@ package unconvert_test
 import (
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/lint/passes/unconvert"
 	"golang.org/x/tools/go/analysis/analysistest"
 )
 
 func Test(t *testing.T) {
+	if testutils.NightlyStress() {
+		t.Skip("Go cache files don't work under stress")
+	}
 	testdata := analysistest.TestData()
 	analysistest.Run(t, testdata, unconvert.Analyzer, "a")
 }

